File: src/main/js/app/reducers/adminReducer.js
const adminReducer = (state = [], action) => {
    switch (action.type) {
        case 'ADMIN_BECOME':
            return {
                ...state,
                isAdmin: true
            };
        case 'ADMIN_STOP':
            return {
                ...state,
                isAdmin: false
            };
        default:
            return state;
    }
};

export default adminReducer

File: API/src/main/java/net/devtales/blog/state/AdminReducer.java
package net.devtales.blog.state;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

public class AdminReducer {
    @JsonProperty("isAdmin")
    private boolean isAdmin;

    @JsonIgnore
    public boolean isAdmin() {
        return isAdmin;
    }

    public AdminReducer setAdmin(boolean isAdmin) {
        this.isAdmin = isAdmin;
        return this;
    }
}
